modules/kvtagger/tests/test_kvtagdb.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2016 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "kvtagdb.h"
+#include "testutils.h"
+#include <stdio.h>
+#include <string.h>
+
+#define KVTAGDB_TESTCASE(testfunc, ...) {testcase_begin("%s(%s)", #testfunc, #__VA_ARGS__); testfunc(__VA_ARGS__); testcase_end();}
+
+static void
+_foreach_ctr(gpointer arg, const TagRecord *record)
+{
+  int *ctr = (int *)arg;
+  ++(*ctr);
+}
+
+static void
+_test_empty_db(KVTagDB *tagdb)
+{
+  assert_false(kvtagdb_is_loaded(tagdb) == TRUE, "Empty KVTagDB should be in unloaded state.");
+  assert_false(kvtagdb_is_indexed(tagdb) == TRUE, "Empty KVTagDB should be in un-indexed state.");
+  assert_false(kvtagdb_contains(tagdb, "selector") == TRUE, "Method kvtagdb_contains should work with empty KVTagDB.");
+  assert_true(kvtagdb_number_of_tags(tagdb, "selector") == 0, "Method kvtagdb_number should work with empty KVTagDB.");
+  int ctr = 0;
+  kvtagdb_foreach_tag(tagdb, "selector", _foreach_ctr, (gpointer)&ctr);
+  assert_true(ctr == 0, "Method kvtagdb_foreach_tag should work for with empty KVTagDB.");
+}
+
+static void
+test_empty_db()
+{
+  KVTagDB *tagdb = kvtagdb_new();
+
+  _test_empty_db(tagdb);
+
+  kvtagdb_unref(tagdb);
+}
+
+static void
+test_purge_empty_db()
+{
+  KVTagDB *tagdb = kvtagdb_new();
+
+  kvtagdb_purge(tagdb);
+  _test_empty_db(tagdb);
+
+  kvtagdb_unref(tagdb);
+}
+
+static void
+test_index_empty_db()
+{
+  KVTagDB *tagdb = kvtagdb_new();
+
+  kvtagdb_index(tagdb);
+  _test_empty_db(tagdb);
+
+  kvtagdb_unref(tagdb);
+}
+
+static void
+_fill_tagdb(KVTagDB *tagdb, const gchar *selector_base,
+            const gchar *name_base, const gchar *value_base,
+            int number_of_selectors, int number_of_nv_pairs_per_selector)
+{
+  int i, j;
+  for (i = 0; i < number_of_selectors; i++)
+    {
+      for (j = 0; j < number_of_nv_pairs_per_selector; j++)
+        {
+          TagRecord record = {
```

Comment:
Eclipse thinks that the parenthesis should start in a new line, but I see this convention a lot. Which one is correct?